crates/miden-objects/src/block/proven_block.rs

```diff
@@ -61,6 +76,7 @@ impl ProvenBlock {
 impl Serializable for ProvenBlock {
     fn write_into<W: ByteWriter>(&self, target: &mut W) {
         self.header.write_into(target);
+        self.signature.write_into(target);
         self.body.write_into(target);
         self.proof.write_into(target);
```

Comment:
Let's change the serialization order to match the order of fields in the struct.